ci/libs/utils.sh

```diff
@@ -0,0 +1,160 @@
+#!/bin/bash
+
+. ci/libs/variables.sh
+
+cleanup(){
+	rm -rf $TMP_DIR/vol*
+	rm -rf $MNT_DIR/logs
+    docker stop $(docker ps -aq)
+    docker rm $(docker ps -aq) 
+}
+
+# start_controller CONTROLLER_IP
+start_controller() {
+	controller_id=$(docker run -d --net stg-net --ip $1\
+        -P --expose 3260 --expose 9501 --expose 9502-9504 $JI \
+        env REPLICATION_FACTOR="$3" launch controller --frontend gotgt\
+        --frontendIP "$1" "$2")
+	echo "$controller_id"
+}
+
+# start_replica CONTROLLER_IP REPLICA_IP folder_name
+start_replica() {
+	replica_id=$(docker run -d -it --net stg-net --ip "$2"\
+        -P --expose 9502-9504 -v $TMP_DIR/"$3":/"$3" $JI \
+        launch replica --frontendIP "$1" --listen "$2":9502\
+        --size 2g /"$3")
+	echo "$replica_id"
+}
+
+collect_logs_and_exit() {
+	echo "--------------------------docker ps -a-------------------------------------"
+	docker ps -a
+
+	echo "--------------------------CONTROLLER REST output---------------------------"
+	curl http://$CONTROLLER_IP:9501/v1/volumes | jq
+	curl http://$CONTROLLER_IP:9501/v1/replicas | jq
+	echo "--------------------------REPLICA 1 LOGS ----------------------------------"
+	curl http://$REPLICA_IP1:9502/v1/replicas | jq
+	echo "--------------------------REPLICA 2 LOGS ----------------------------------"
+	curl http://$REPLICA_IP2:9502/v1/replicas | jq
+	echo "--------------------------REPLICA 3  LOGS ---------------------------------"
+	curl http://$REPLICA_IP3:9502/v1/replicas | jq
+
+	#Take system output
+	ps -auxwww
+	top -n 10 -b
+	netstat -nap
+
+	echo "ls VOL1>>"
+	ls -ltr $TMP_DIR/vol1/
+	echo "ls VOL2>>"
+	ls -ltr $TMP_DIR/vol2/
+	echo "ls VOL3>>"
+	ls -ltr $TMP_DIR/vol3/
+	#Below is to get stack traces of longhorn processes
+	kill -SIGABRT $(ps -auxwww | grep -w longhorn | grep -v grep | awk '{print $2}')
+
+	echo "--------------------------ORIGINAL CONTROLLER LOGS ------------------------"
```

Comment:
should we pipe to a tail command so as to get like relevant lines?

Reply:
sorry, i didn't understand what you meant :(

Reply:
docker logs are huge lines, so I am trying to say `docker logs <container> | tail -10` or something like that for ease of understanding.

Reply:
We need entire logs to trace back the error from beginning